test/mlir/test/gc/Transforms/cpu-phyaical-register.mlir

```diff
@@ -0,0 +1,666 @@
+  return %2 : tensor<11008x4096xf32>
+}
+
+// CHECK-LABEL: func @reduce_keepdimtest1
+// CHECK: %[[C32:.*]] = arith.constant 32 : index
+// CHECK: %[[C64:.*]] = arith.constant 64 : index
+// CHECK: %[[C16:.*]] = arith.constant 16 : index
+// CHECK: %[[C1:.*]] = arith.constant 1 : index
+// CHECK: %[[C0:.*]] = arith.constant 0 : index
+// CHECK: %[[CST:.*]] = arith.constant 0.000000e+00 : f32
+// CHECK: %[[EMPTY0:.*]] = tensor.empty() : tensor<16x64xf32>
+// CHECK: %[[EMPTY1:.*]] = tensor.empty() : tensor<16x1x64xf32>
+// CHECK: scf.for
+// CHECK: scf.for
+// CHECK: %[[READ0:.*]] = vector.transfer_read {{.*}}, {{.*}} {in_bounds = [true]} : tensor<16x64xf32>, vector<16xf32>
+// CHECK: scf.for
+// CHECK: %[[READ1:.*]] = vector.transfer_read {{.*}}, {{.*}} {in_bounds = [true]} : tensor<16x32x64xf32>, vector<16xf32>
+// CHECK: %[[ADD0:.*]] = arith.addf %[[READ1]],  {{.*}} : vector<16xf32>
+// CHECK: scf.yield
+// CHECK: %[[WRITE:.*]] = vector.transfer_write {{.*}}, {{.*}} {in_bounds = [true]} : vector<16xf32>, tensor<16x64xf32>
+// CHECK: scf.yield
+// CHECK: scf.for %[[arg1:.*]] = %[[C0]] to %[[C16]] step %[[C1]] iter_args(%[[arg2:.*]] = %[[EMPTY1]]) -> (tensor<16x1x64xf32>)
+// CHECK: scf.for %[[arg3:.*]] = %[[C0]] to %[[C1]] step %[[C1]] iter_args(%[[arg4:.*]] = %[[arg2]]) -> (tensor<16x1x64xf32>)
+// CHECK: scf.for %[[arg5:.*]] = %[[C0]] to %[[C64]] step %[[C16]] iter_args(%[[arg6:.*]] = %[[arg4]]) -> (tensor<16x1x64xf32>)
+// CHECK: %[[APPLY0:.*]] = affine.apply #[[map0]]()[%[[arg3]], %[[arg5]]]
+// CHECK: %[[READ2:.*]] = vector.transfer_read {{.*}}, {{.*}} {in_bounds = [true]} : tensor<16x64xf32>, vector<16xf32>
+// CHECK: %[[WRITE1:.*]] = vector.transfer_write {{.*}}, {{.*}} {in_bounds = [true]} : vector<16xf32>, tensor<16x1x64xf32>
+func.func @reduce_keepdimtest1(%arg0: tensor<16x32x64xf32>) -> tensor<16x1x64xf32> {
+  %0 = tensor.empty() : tensor<16x64xf32>
+  %reduce = linalg.reduce
+      ins(%arg0:tensor<16x32x64xf32>)
+      outs(%0:tensor<16x64xf32>)
+      dimensions = [1]
+      (%in: f32, %out: f32) {
+        %1 = arith.addf %out, %in: f32
+        linalg.yield %1: f32
+      }
+  %2 = tensor.expand_shape %reduce [[0],[1, 2]] output_shape [16, 1, 64] : tensor<16x64xf32> into tensor<16x1x64xf32>
+  return %2 : tensor<16x1x64xf32>
+}
+
+// CHECK-LABEL: func @fc_relu_test2
+// CHECK: %[[MATMUL:.*]] = linalg.matmul
+// CHECK: scf.for
+// CHECK: scf.for
+// CHECK: %[[READ0:.*]] = vector.transfer_read {{.*}}, {{.*}}: tensor<512x512xf32>, vector<16xf32>
+// CHECK: %[[READ1:.*]] = vector.transfer_read {{.*}}, {{.*}}: tensor<512x512xf32>, vector<16xf32>
+// CHECK:  %[[ADD0:.*]] = arith.addf %[[READ1]], %[[READ0]] : vector<16xf32>
+// CHECK:  %[[ADD1:.*]] = arith.maximumf %[[ADD0]], {{.*}} : vector<16xf32>
+// CHECK: %[[WRITE:.*]] = vector.transfer_write {{.*}}, {{.*}} : vector<16xf32>, tensor<512x512xf32>
+func.func @fc_relu_test2(%lhs: tensor<512x512xf32>, %rhs: tensor<512x512xf32>,
```

Comment:
How do you ensure the correctness of the transformed IR? Is there any accuracy test that covers this pass?

Reply:
We now rely on CI to test the correctness of each op, and MLP pattern to test the correctness of the pattern results.

Reply:
The tests in this file usually contain multiple ops. Do you mean that the MLP patterns can cover the IRs in this file?

Reply:
I understand. The purpose of this test file is just to test whether the conversion to the target IR can be done. It does not guarantee the correctness of the calculation results. To ensure the correctness, we need to test it through CI.